mlir/lib/Dialect/Rock/utility/fusionUtils.cpp

```diff
@@ -0,0 +1,79 @@
+//===- fusionUtils.cpp - Rock utility for fusion -----------------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------===//
+
+#include "mlir/Dialect/Rock/utility/fusionUtils.h"
+#include "mlir/Analysis/BufferDependencyAnalysis.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Rock/IR/Rock.h"
+#include "mlir/Dialect/Rock/Tuning/GridwiseGemmParams.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/Pass/AnalysisManager.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Support/LogicalResult.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Support/Casting.h"
+
+using namespace mlir;
+using namespace mlir::rock;
+
+std::optional<memref::AllocOp> mlir::rock::getAllocation(Value value) {
```

Comment:
This looks very similar to this:
https://github.com/ROCm/rocMLIR/blob/762fdc80e1be85c4124ff5abe827fd8aed5c0728/mlir/lib/Dialect/Rock/utility/loweringUtils.cpp#L519
Would it be possible to unify them?

Reply:
Thanks @giuseros . It is possible if you are ok to 1) template the function and 2) move the implementation to the header.  Do you think it is going to be ok?

Reply:
Hi @giuseros, I managed to address your comment. Can you, please, re-review?

Reply:
Ah right, you are looking for memref allocs and I was looking for gpuallocs. I don't have strong opinions, so feel free to do it if you like , but I won't stop the approval :)

Reply:
I managed to combine the implementations of both of them.
done